### DISCUSSION TITLE
Question about compile to Static Lib.

### DISCUSSION THREAD

**Discussion by Doubl3MD** [halide/Halide#7089] [Q&A]
I'm building a Halide pipeline and it is multistage.
But the multistage is achieved by creating multiple callable functions from a halide (for now multiple)compiled static lib.

So to make it as clear as possible about what I try to achieve:

I make a Halide accelerated function and compile to static. 
I make a second Halide accelerated function and compile it to a second static lib.
But when i try to include and run both functions from the lib i run into 'multiple defined symbols found'.
This, i guess, is because halide compiles every func with the same stuff to a lib.
In my case resulting in 2 lib files and 2 headers and so causing the symbol issue because there is some kind of 'symbol overlap' in both libs.

My question is : can i compile multiple halide funcs to 1 static library? Thus avoiding the symbol issues.

Or is there a more conventional way to get what i want to achieve?

Another more practical example just in case it was not clear what i meant/want:  Create a func that scales and another one that blurs, and another one that pimps or whatever.... compile all AOT (static lib.. the way to go?). And all can be called separately and in no particular order.

Thanks for any hint in the right direction.

**Comment by Doubl3MD** [comment]
I'm looking into Halide::Pipeline...

**Comment by shoaibkamil** [comment]
Take a look at lesson 15 in the tutorial.  By default, a Halide pipeline will contain a copy of the Halide runtime unless the target includes `no_runtime`.  So for multiple pipelines going into the same compilation unit, you have two options:

1) Generate one static library without specifying `no_runtime` in the target, and then generate the rest using `no_runtime`, or

2) Generate all the static libraries using `no_runtime` in the target, then generate a separate runtime library.
